### PR TITLE
Fix backup check

### LINKED ISSUE #155: Enhance backup check
We need to do a better job checking that one backup is present [here](https://git.io/vSYxR)

### PR DESCRIPTION
**Pull request by johnpmitsch**
Fixes #155
We need way of checking that one and only one backup is present.

### CODE REVIEW

**Comment by sthirugn** [comment]
Are you sure you don't want to solve this with relatively simple one line of code?

```sh
  when: (online_backup and standard_backup) or (online_backup and rhel_migration_backup) or (standard_backup and rhel_migration_backup)
```

**Comment by johnpmitsch** [comment]
@sthirugn if we can use a module it will be more readable (IMO) and easier to introduce new scenarios to, but I'll def keep that in mind as a backup solution

**Comment by sthirugn** [comment]
testing this with a test stub:
- name: Set all backup types
  set_fact:
    online_backup: true
    standard_backup: false
    rhel_migration_backup: false

I get this error:
TASK [satellite-clone : Check only one backup is present] **********************
Tuesday 28 March 2017  15:15:38 -0400 (0:00:00.014)       0:00:01.747 ********* 
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Boolean rhel_migration_backup not in either boolean list"}

**Comment by johnpmitsch** [comment]
@sthirugn ah sorry, still working on it but pushed it up so I could pull it on a machine, let me mark it WIP

File: roles/satellite-clone/tasks/backup_check.yml [hunk 1]
```diff
@@ -95,4 +95,4 @@
       remove some files so there is only one backup present. For an online backup, you need
       candlepin.dump, foreman.dump, and mongo_dump folder. For a standard (offline) backup,
       you will need mongo_data.tar.gz and pgsql_data.tar.gz.
```
Comment (sthirugn) [inline roles/satellite-clone/tasks/backup_check.yml]: I liked the msg from your previous commit though which detailed all three scenarios clearly.

Comment (johnpmitsch) [inline roles/satellite-clone/tasks/backup_check.yml]: @sthirugn :+1: will add back

File: roles/satellite-clone/tasks/backup_check.yml [hunk 2]
```diff
@@ -91,8 +91,10 @@
...
     msg: |
-      It looks like there is both an online and standard backup present. You will have to
-      remove some files so there is only one backup present. For an online backup, you need
-      candlepin.dump, foreman.dump, and mongo_dump folder. For a standard (offline) backup,
-      you will need mongo_data.tar.gz and pgsql_data.tar.gz.
-  when: online_backup and standard_backup and not rhel_migration
+      It looks like there is more than one type of backup present. You will have to
+      remove some files so there is only one backup present. here are the backup files
+      for each scenario.
+      Standard backup scenario - config_files.tar.gz, mongo_data.tar.gz, pgsql_data.tar.gz, (optional) pulp_data.tar
```
Comment (sthirugn) [inline roles/satellite-clone/tasks/backup_check.yml]: nitpick:
s/here are the backup files for each scenario./Here are the required backup files for each scenario.

**Comment by johnpmitsch** [comment]
@sthirugn updated
Here are the results from dolly:
[62_rhel7]
successful clone, repo sync works fine 

[61_rhel7] 
successful clone, repo sync works fine 

[62_rhel6_migration]
successful clone, repo sync works fine 

[62_rhel6_migration_invalid]
failed on backup check (expected)

[62_online_with_pulp_data]
successful clone, repo sync works fine  

[62_online_without_pulp_data]
Installer failed , there was an error about a timeout during db:seed,
this doesn't seem to be on us, I've ran this scenario plenty of times recently without issue.
A re-run of the installer ran fine, and after re-running the playbook, I had a functioning
Satellite with content (repo sync worked fine)

**Comment by sthirugn** [comment]
ACK

ty @johnpmitsch

**Pull request merged**